neo4j/datadog_checks/neo4j/neo4j.py

```diff
@@ -1,167 +1,304 @@
-        'locks': 'neo4j.locks',
-        'numberofaverteddeadlocks': 'neo4j.adverted.locks',
-        'numberofrelationshipidsinuse': 'neo4j.relationship.ids.inuse',
-        'numberofpropertyidsinuse': 'neo4j.property.ids.inuse',
-        'numberofnodeidsinuse': 'neo4j.node.ids.inuse',
-        'numberofrelationshiptypeidsinuse': 'neo4j.relationshiptype.ids.inuse',
-        'memorypools': 'neo4j.memory.pools',
-        'pins': 'neo4j.page.cache.pins',
-        'evictions': 'neo4j.page.cache.evictions',
-        'byteswritten': 'neo4j.bytes.written',
-        'filemappings': 'neo4j.page.cache.file.mappings',
-        'fileunmappings': 'neo4j.page.cache.file.unmappings',
-        'bytesread': 'neo4j.bytes.read',
-        'flushes': 'neo4j.page.cache.flushes',
-        'evictionexceptions': 'neo4j.page.cache.eviction.exceptions',
-        'faults': 'neo4j.page.cache.faults',
-        'ha.pull_interval': 'neo4j.ha.pull_interval',
-        'dbms.memory.pagecache.size': 'neo4j.dbms.memory.pagecache.size',
-    }
-
-    def check(self, _):
-        host, port, server_name = self._get_config(self.instance)
-        tags = self.instance.get('tags', [])
-        tags.append('server_name:{}'.format(server_name))
-        service_check_tags = tags + ['url:{}'.format(host)]
-
-        # Neo specific
-        # Create payload using built-in Neo4j queryJmx stored procedure
-        payload = {
-            "statements": [
-                {
-                    "statement": "CALL dbms.queryJmx('org.neo4j:*') yield attributes with  "
-                    "keys(attributes) as k, attributes unwind k as "
-                    "row return row, attributes[row]['value'];"
-                }
-            ]
-        }
-        try:
-            version = self._get_version(host, port, service_check_tags)
+from datadog_checks.base import ConfigurationError
+from datadog_checks.base.checks.prometheus.prometheus_base import PrometheusCheck
+
+GLOBAL_DB_NAME = 'global'
+NAMESPACE = 'neo4j'
+
+
+class Config:
+    def __init__(self, host, port, neo4j_version, neo4j_dbs, exclude_labels, instance_tags):
+        self.host = host
+        self.port = port
+        self.neo4j_version = neo4j_version
+        self.neo4j_dbs = neo4j_dbs
+        self.exclude_labels = exclude_labels
+        self.instance_tags = instance_tags
+
+    def __eq__(self, other):
+        return (
+            self.host == other.host
+            and self.port == other.port
+            and self.neo4j_version == self.neo4j_version
+            and self.neo4j_dbs == other.neo4j_dbs
+            and self.exclude_labels == other.exclude_labels
+            and self.instance_tags == other.instance_tags
+        )
+
+
+class Neo4jCheck(PrometheusCheck):
+    def check(self, instance):
+        self._set_whitelisted_metrics()
+        config = self._get_config(instance=instance)
+        self.exclude_labels = config.exclude_labels
+
+        endpoint = 'http://{}:{}/metrics'.format(config.host, config.port)
```

Comment:
FYI: You don't provide the ability to use https here.

Reply:
Added in ability to use https.